overlord/servicestate/servicemgr.go

```diff
@@ -60,3 +194,123 @@ func serviceControlAffectedSnaps(t *state.Task) ([]string, error) {
 	}
 	return []string{serviceAction.SnapName}, nil
 }
+
+func restartServicesKilledInSnapdSnapRefresh(modified map[*snap.Info][]*snap.AppInfo) error {
+	// we decide on which services to restart by identifying (out of the set of
+	// services we just modified) services that were stopped after
+	// usr-lib-snapd.mount was written, but before usr-lib-snapd.mount was last
+	// stopped - this is the time window in which snapd (accidentally) killed
+	// all snap services using Requires=, see LP #1924805 for full details, so
+	// we need to undo that by restarting those snaps
+
+	st, err := os.Stat(filepath.Join(dirs.SnapServicesDir, wrappers.SnapdToolingMountUnit))
+	if err != nil {
+		return err
+	}
+
+	// TODO: we should check if usr-lib-snapd.mount was modified before the
+	// current boot time, if it was then we can just skip this since we know
+	// any service stops that happened were unrelated
```

Comment:
doing this would be good, it shouldn't be too hard?

Reply:
this is done now